src/rpc/server.zig

```diff
@@ -0,0 +1,547 @@
+        fmt: MethodFmt,
+        comptime fmt_str: []const u8,
+        fmt_options: std.fmt.FormatOptions,
+        writer: anytype,
+    ) @TypeOf(writer).Error!void {
+        _ = fmt_options;
+        if (fmt_str.len != 0) std.fmt.invalidFmtError(fmt_str, fmt);
+        try fmt.method.write(writer);
+    }
+};
+
+test Server {
+    const allocator = std.testing.allocator;
+
+    var prng = std.Random.DefaultPrng.init(0);
+    const random = prng.random();
+
+    // const logger: sig.trace.Logger = .{ .direct_print = .{ .max_level = .trace } };
+    const logger: sig.trace.Logger = .noop;
+
+    var test_data_dir = try std.fs.cwd().openDir("data/test-data", .{ .iterate = true });
+    defer test_data_dir.close();
+
+    var tmp_dir_root = std.testing.tmpDir(.{});
+    defer tmp_dir_root.cleanup();
+    const tmp_dir = tmp_dir_root.dir;
+
+    var snap_dir = try tmp_dir.makeOpenPath("snapshot", .{ .iterate = true });
+    defer snap_dir.close();
+
+    const SnapshotFiles = sig.accounts_db.snapshots.SnapshotFiles;
+    const snap_files = try SnapshotFiles.find(allocator, test_data_dir);
+
+    const full_snap_name_bounded = snap_files.full_snapshot.snapshotNameStr();
+    const maybe_inc_snap_name_bounded = if (snap_files.incremental_snapshot) |inc|
+        inc.snapshotNameStr()
+    else
+        null;
+
+    {
+        const full_snap_name = full_snap_name_bounded.constSlice();
+
+        try test_data_dir.copyFile(full_snap_name, snap_dir, full_snap_name, .{});
+        const full_snap_file = try snap_dir.openFile(full_snap_name, .{});
+        defer full_snap_file.close();
+
+        const unpack = sig.accounts_db.snapshots.parallelUnpackZstdTarBall;
+        try unpack(allocator, logger, full_snap_file, snap_dir, 1, true);
+    }
+
+    if (maybe_inc_snap_name_bounded) |inc_snap_name_bounded| {
+        const inc_snap_name = inc_snap_name_bounded.constSlice();
+
+        try test_data_dir.copyFile(inc_snap_name, snap_dir, inc_snap_name, .{});
+        const inc_snap_file = try snap_dir.openFile(inc_snap_name, .{});
+        defer inc_snap_file.close();
+
+        const unpack = sig.accounts_db.snapshots.parallelUnpackZstdTarBall;
+        try unpack(allocator, logger, inc_snap_file, snap_dir, 1, false);
+    }
+
+    const AllSnapshotFields = sig.accounts_db.snapshots.AllSnapshotFields;
+    var all_snap_fields = try AllSnapshotFields.fromFiles(
+        allocator,
+        logger,
+        snap_dir,
+        snap_files,
+    );
+    defer all_snap_fields.deinit(allocator);
+
+    var accountsdb = try sig.accounts_db.AccountsDB.init(.{
+        .allocator = allocator,
+        .logger = logger,
+        .snapshot_dir = snap_dir,
+        .geyser_writer = null,
+        .gossip_view = null,
+        .index_allocation = .ram,
+        .number_of_index_shards = 4,
+        .lru_size = null,
+    });
+    defer accountsdb.deinit();
+    _ = try accountsdb.loadWithDefaults(allocator, &all_snap_fields, 1, true, 1500, false, false);
```

Comment:
1500 is overkill here, should change to 300

Reply:
To be clear I was just using the value used by most other sites I saw; will change it to 300.

Reply:
The other instance of this is in [accountsdb itself](https://github.com/Syndica/sig/blob/82a241843c3d1ba27c898d32f2ac27741f9dabe4/src/accountsdb/db.zig#L4198); should I change that too?

Reply:
yeah should change that instance too - missed that - thanks

Reply:
78fc6dec173897d993bed8c9d82b6808aeaf8b3d
Also changed it in the fuzz testing code